File: Templates/sliding_window_variable_shortest.cpp
#include <vector>
#include <algorithm>
#include <climits>

// Placeholder for user-defined helpers
bool valid(const std::vector<int>& window);
int optimal(const std::vector<int>& window);

int sliding_window_flexible_shortest(const std::vector<int>& input) {
    std::vector<int> window;
    int answer = INT_MAX;
    int left = 0;
    for (int right = 0; right < (int)input.size(); ++right) {
        window.push_back(input[right]);
        while (valid(window)) {
            answer = std::min(answer, optimal(window));
            int left_elem = input[left];
            auto it = std::find(window.begin(), window.end(), left_elem);
            if (it != window.end()) {
                window.erase(it);
            }
            ++left;
        }
    }
    return answer;
}

File: Templates/sliding_window_variable_shortest.py
def sliding_window_flexible_shortest(input):
    window = []
    anwser = 0
    left = 0
    for right in range(len(input)):
        window.append(input[right])
        while valid(window):
            anwser = min(anwser, window)      # window is guaranteed to be valid here
            window.remove(input[left])
            left += 1
    return anwser
